src/Engines/WebGPU/webgpuShaderProcessorsGLSL.ts

```diff
@@ -0,0 +1,319 @@
+import { Nullable } from '../../types';
+import { IShaderProcessor } from '../Processors/iShaderProcessor';
+import { ShaderProcessingContext } from "../Processors/shaderProcessingOptions";
+import { WebGPUShaderProcessingContext, WebGPUBufferDescription } from './webgpuShaderProcessingContext';
+import * as WebGPUConstants from './webgpuConstants';
+import { Logger } from '../../Misc/logger';
+import { ThinEngine } from "../thinEngine";
+import { WebGPUShaderProcessor } from "./webgpuShaderProcessor";
+import { ShaderLanguage } from "../../Materials/shaderLanguage";
+
+/** @hidden */
+export class WebGPUShaderProcessorGLSL extends WebGPUShaderProcessor implements IShaderProcessor {
+
+    protected _missingVaryings: Array<string> = [];
+    protected _textureArrayProcessing: Array<string> = [];
+    protected _preProcessors: { [key: string]: string };
+
+    public shaderLanguage = ShaderLanguage.GLSL;
+
+    protected _getArraySize(name: string, type: string, preProcessors: { [key: string]: string }): [string, string, number] {
+        let length = 0;
+        const startArray = name.indexOf("[");
+        const endArray = name.indexOf("]");
+        if (startArray > 0 && endArray > 0) {
+            const lengthInString = name.substring(startArray + 1, endArray);
+            length = +(lengthInString);
+            if (isNaN(length)) {
+                length = +(preProcessors[lengthInString.trim()]);
+            }
+            name = name.substr(0, startArray);
+        }
+        return [name, type, length];
+    }
+
+    public initializeShaders(processingContext: Nullable<ShaderProcessingContext>): void {
+        this.webgpuProcessingContext = processingContext as WebGPUShaderProcessingContext;
+
+        this._missingVaryings.length = 0;
+        this._textureArrayProcessing.length = 0;
+    }
+
+    public varyingProcessor(varying: string, isFragment: boolean, preProcessors: { [key: string]: string }, processingContext: Nullable<ShaderProcessingContext>) {
+        this._preProcessors = preProcessors;
+
+        const varyingRegex = new RegExp(/\s*varying\s+(?:(?:highp)?|(?:lowp)?)\s*(\S+)\s+(\S+)\s*;/gm);

```

Comment:
no need to create a new RegExp if you are already using `/.../`
```suggestion
        const varyingRegex = /\s*varying\s+(?:(?:highp)?|(?:lowp)?)\s*(\S+)\s+(\S+)\s*;/gm;
```